### PR TITLE
feat(@angular-ru/common): refactor amountFormat

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
-   [x] Tests for the changes have been added (for bug fixes / features)
-   [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

**Review by mansurishmuhametov — APPROVED** [review]

**Review by nikitasadchikov — APPROVED** [review]

**Review by Schirbak — APPROVED** [review]

File: packages/common/directives/src/amount-format/amount-format.directive.ts [hunk 1]
```diff
@@ -1,59 +1,286 @@
...
+        this.ngControl?.reset('');
+        this.element.value = '-';
+    }
+
+    private getFractionSeparator(): string {
+        return getFractionSeparator(this.options.lang ?? DEFAULT_AMOUNT_OPTIONS.lang);
+    }
+
+    private removeNonNumericSymbols(): string {
+        return this.element.value === '-' ? this.element.value : removeNonNumericSymbols(this.element.value);
```
Comment (splincode) [inline packages/common/directives/src/amount-format/amount-format.directive.ts:246]: `this.viewModelIsMinus()`

File: packages/common/directives/src/amount-format/amount-format.directive.ts [hunk 2]
```diff
@@ -1,59 +1,286 @@
...
+
+    private setSelectionRangeBy(safePosition: boolean, value: string): void {
+        try {
+            if (safePosition) {
+                this.element.setSelectionRange(this.cursorPointer, this.cursorPointer);
+            } else {
+                this.cursorPointer = value.length;
+                this.element.setSelectionRange(this.cursorPointer, this.cursorPointer);
+            }
+        } catch {}
```
Comment (splincode) [inline packages/common/directives/src/amount-format/amount-format.directive.ts:265]: need add note

File: packages/common/integration/tests/directives/amount-format.spec.ts [hunk 3]
```diff
@@ -1,113 +1,1104 @@
...
+            };
+
+            // MOCK
+            directive = new AmountFormatDirective(element as ElementRef, DEFAULT_AMOUNT_OPTIONS, control as NgControl);
+        });
+
+        describe('ru-RU', () => {
+            describe('non numeric by ru-RU', () => {
+                it('ngModel value is empty by ru-RU', () => {
+                    directive.format();
```
Comment (Schirbak) [inline packages/common/integration/tests/directives/amount-format.spec.ts:45]: seriously, it's a crazy big job to match it all

Comment (Schirbak) [reply packages/common/integration/tests/directives/amount-format.spec.ts:45]: about syntax: you deliberately avoided the `test.each` construct?

**Review by Schirbak — COMMENTED** [review]
great work!
i have some suggestions to rename

File: packages/common/directives/src/amount-format/amount-format.directive.ts [hunk 4]
```diff
@@ -1,59 +1,290 @@
...
+    public ngAfterViewInit(): void {
         this.format();
     }
 
-    @HostListener('blur')
+    public ngOnDestroy(): void {
+        this.unlisten();
+    }
+
+    public listen(): void {
```
[the remark this answers is not in the document]
Comment (splincode) [reply packages/common/directives/src/amount-format/amount-format.directive.ts:87]: maybe we can create common interface for similar cases?

File: packages/common/integration/tests/directives/amount-format.spec.ts (hunk 3 above)
Comment (splincode) [reply packages/common/integration/tests/directives/amount-format.spec.ts:45]: `about syntax: you deliberately avoided the test.each construct?` 

yes, for better readability of tests

File: packages/common/directives/src/amount-format/amount-format.directive.ts [hunk 5]
```diff
@@ -1,59 +1,290 @@
...
         }
 
-        const stringValue: string = this.value.replace(/,/g, '.').replace(/(?!-)[^0-9.]/g, '');
-        const numberValue: number = parseFloat(stringValue);
+        const numberValue: number = this.getNumberValueWithGaussRounded();
+        this.setModelValueByGaussRounded(numberValue);
+
+        const stringValue: string = this.prepareConvertedToLocaleValue(numberValue, fraction);
+
+        this.element.value = isNaN(this.ngControl?.value) ? '' : stringValue;
```
[the remark this answers is not in the document]
Comment (splincode) [reply packages/common/directives/src/amount-format/amount-format.directive.ts:137]: add note

File: packages/common/directives/src/amount-format/amount-format.directive.ts [hunk 6]
```diff
@@ -1,59 +1,290 @@
...
 
-    @HostListener('blur')
+    public ngOnDestroy(): void {
+        this.unlisten();
+    }
+
+    public listen(): void {
+        this.ngZone?.runOutsideAngular((): void => {
+            this.subscriptions.add(
+                fromEvent(this.element, 'input').subscribe((): void => {
```
[the remark this answers is not in the document]
Comment (Schirbak) [reply packages/common/directives/src/amount-format/amount-format.directive.ts:90]: ok, host listener work inside the zone

**Review by Schirbak — APPROVED** [review]

**Review by nikitasadchikov — APPROVED** [review]

**Pull request merged**